swiftui-project/View/Login/LoginViewModel.swift

```diff
@@ -41,4 +46,28 @@ class LoginViewModel: ObservableObject {
         return self.email.isEmpty || self.password.isEmpty
     }
     
+    func login() {
+        Task {
+            do {
+                self.isLoggingIn = true
+                self.validateLogin()
+                
+                if areCredentialsValid {
+                    let request = LoginRequest(email: self.email, password: self.password)
+                    let loginResponse = try await repository.login(request: request)
+                    let jwtToken = loginResponse.jwt
+                    // attach the jwtToken
+                    let networkManager = NetworkManager(jwtToken: jwtToken)
```

Comment:
Here you are creating a brand new local instance of `NetworkManager` with that jwtToken. The `NetworkManager` that you're using in the repository will be a completely different instance that won't be initialized with that jwtToken.

You should inject the `NetworkManager` here in the `LoginViewModel` and set its jwtToken on a shared singleton instance instead for this to work but it's not a nice solution as you're exposing the `NetworkManager` to some random view model.

But even better (easy solution) - make a storage class of some sorts (i.e. AuthStorage/JWTStorage/etc.) and have it contain the `jwtToken` property. Define that class in the Factory's Container extension as you have done for the NetworkManager and instead of injecting the network manager here, inject it instead, setting the jwtToken to it. Then inject it in the JWTInterceptor class of yours as well and make the JWT interceptor adapt take the value from the singleton storage class instance.

I suggest you try it the above way and if you happen to run into some spare time of yours and curiosity you may check out the method described below:

(Optional) Even better (tougher solution), you can research how to store and retrieve data from the keychain. There are two types of app storage that we often use in mobile apps - AppStorage/UserDefaults and Keychain. The first one we'll talk about in a next lecture but unfortunately we won't have time to play around with the latter. It's the most niche and secure storage of the two and it's perfect for sensitive data that has to be persisted such as a JWT token.

Reply:
Problem with the token solved